Foreman/DataCache.cs

```diff
@@ -1281,9 +1210,42 @@ private static void InterpretLuaRecipe(String name, LuaTable values)
 
                 if (name == null)
                     name = results.ElementAt(0).Key.Name;
+
+                // check if recipe gets researched
+                bool? recipeEnabled = GetLuaValue<bool?>(timeSource, "enabled", true); // nullable, because some starting recipes don't have enabled flag
+                if (recipeEnabled.HasValue && recipeEnabled == false) // needs to be researched
+                {
+                    bool found = false;
+                    System.Collections.IDictionaryEnumerator techTableEnum = techTable.GetEnumerator();
+                    while (techTableEnum.MoveNext())
+                    {
+                        bool? techEnabled = GetLuaValue<bool?>(techTableEnum.Value as LuaTable, "enabled", true);
+                        if (!techEnabled.HasValue || techEnabled == true)

```

Comment:
I'd heavily encourage using early breaks/returns here it should remove almost half the nesting.

Reply:
done